subsys/fs/fat_fs.c

```diff
@@ -463,9 +481,17 @@ static int fatfs_mount(struct fs_mount_t *mountp)
 static int fatfs_unmount(struct fs_mount_t *mountp)
 {
 	FRESULT res;
+	DRESULT disk_res;
+	uint8_t param = 0;
 
 	res = f_mount(NULL, translate_path(mountp->mnt_point), 0);
 
+	/* Make direct disk IOCTL call to deinit disk */
+	disk_res = disk_ioctl(((FATFS *)mountp->fs_data)->pdrv, CTRL_POWER, &param);
+	if (disk_res != RES_OK) {
+		return translate_disk_error(disk_res);
+	}
+
```

Comment:
There should be some logging here. Otherwise how do you know whether this one failed or the one below if they now share common errno space?

Reply:
Good catch- I'll add this on next push, once we've determined how to enforce the requirements around INIT/DEINIT IOCTLS